src/machine/machine_rp2_i2c.go

```diff
@@ -218,8 +215,8 @@ func (i2c *I2C) enable() {
 //
 //go:inline
 func (i2c *I2C) disable() error {
-	const MAX_T_POLL_COUNT = 64 // 64 us timeout corresponds to around 1000kb/s i2c transfer rate.
-	deadline := ticks() + MAX_T_POLL_COUNT
+	const timeout_us = 40000
```

Comment:
Why did you change this line? This loop does not call gosched

Reply:
Did it have a noticeable effect? What baud are you running?

Reply:
I run i2c at 400kHz.

I had to bump the `disable` deadline because of delays caused by a long interrupt handler. Unlike `gosched`, interrupt handlers can reasonably be assumed to complete in reasonable time (<40ms), but not in (in this case) 64us.

Reply:
I have some reservations jumping the disable timeout by factor >1000 than what the reference implementation suggests. If a interrup handler runs in over a few ms i believe that to be a real issue for tinygo which usually encourages really short interrupts to not negatively interfere with the runtime

Reply:
Can you try 400us and 4ms and see if any of those work?

Reply:
I changed the timeouts to 4ms; my interrupt handler completes in ~1.5ms. Long interrupts are not great, but what else can be done about real-time requirements (e.g. interrupt "bottom halves")? TinyGo doesn't offer goroutine priorities.

Just for my curiosity, where do you see the disable deadline in the pico-sdk reference? I couldn't see it in https://github.com/raspberrypi/pico-sdk/blob/master/src/rp2_common/hardware_i2c/i2c.c.

Reply:
![image](https://github.com/user-attachments/assets/763eaa3b-f557-4ccc-a7b8-2192bd29540f)
Currently on vacations so cant spend too much time right now. Looks like I diverted from the Pico SDK and took the datasheet at its word here.  Prolly get back to this by mid week

Reply:
Thank you for the reference.

> Increment POLL_COUNT by one. If POLL_COUNT >= MAX_T_POLL_COUNT, exit with [...] error [...]

reads to me that MAX_T_POLL_COUNT is a loop counter, not a deadline. And 

> Define a maximum time-out paramater, MAX_T_POLL_COUNT

reads to me as the value being implementation defined. Only the loop iteration delay, `t_i2c_poll`, is defined in terms of bus speed.

Am I reading the datasheet wrong here?

Reply:
You are correct! Looks like I slipped on the naming